jira_completion.py

```diff
@@ -0,0 +1,138 @@
+"""
+Sublime Text 3 Plugin to autocomplete JIRA issues numbers
+by ticket titles.
+
+How to use:
+
+Type "JI" with an optional filter, e.g. "JIauthgoogle", press Alt+/.
+Completion menu should pop up.
+
+How to set up:
+1. Add your username and password to ~/.netrc, e.g.:
+
+    machine jira.iponweb.net login vpetrov password topsecretpassword
+
+CAUTION: async requests are not supported, so your editor will freeze
+until reply from JIRA is received.
+Also, minimal error handling, sorry.
+"""
+
+import base64
+import datetime
+import json
+import netrc
+import sublime
+import sublime_plugin
+import urllib
+
+from urllib.error import HTTPError
+
+
+class Settings:
+    def __init__(self, settings):
+        self._settings = settings
+
+    @property
+    def hostname(self):
+        return self._settings.get('hostname', 'jira.iponweb.net')
+
+    @property
+    def trigger(self):
+        return self._settings.get('trigger', 'JI')
+
+    @property
+    def jira_completion_url(self):
+        return self._settings.get(
+            'jira_completion_url',
+            'https://jira.iponweb.net/rest/api/2/search?'
+            'jql=(watcher+in+({username})+'
+                'OR+assignee+in+({username})+'
+                'OR+reporter+in+({username}))+'
+                'ORDER+BY+updated+DESC,+created+DESC'
+                '&maxResults=100&fields=id,key,summary'
+        )
+
+    @property
+    def timeout(self):
+        return self._settings.get('timeout', 5)
+
+    @property
+    def log_filename(self):
+        return self._settings.get('log_filename', None)
+
+
+def load_settings():
+    return Settings(sublime.load_settings('timesheets.sublime-settings'))
+
+
+class JiraCompleter:
+    def __init__(self, settings):
+        self._settings = settings
+
+    def _log(self, message):
```

Comment:
There is built-in console in Sublime, press Ctrl+~ to summon it. What about write log into this console and don't create log file?